src/v/storage/segment_appender.h

```diff
@@ -73,9 +86,36 @@ class segment_appender {
         return _committed_offset + _bytes_flush_pending;
     }
 
+    /**
+     * @brief Appends a batch.
+     *
+     * See the class doc for the general semantics of append() and the
+     * return value.
+     */
     ss::future<> append(const model::record_batch& batch);
+
+    /**
+     * @brief Appends a buffer of size n starting at buf.
+     *
+     * See the class doc for the general semantics of append() and the
+     * return value.
+     */
     ss::future<> append(const char* buf, const size_t n);
+
+    /**
+     * @brief Appends a byte view.
+     *
+     * See the class doc for the general semantics of append() and the
+     * return value.
+     */
     ss::future<> append(bytes_view s);
+
+    /**
+     * @brief Appends the contents of an iobuf.
+     *
+     * See the class doc for the general semantics of append() and the
+     * return value.
+     */
```

Comment:
These comments are mostly all added so that I can refer readers back to the class doc for the importnat note on append() concurrency since I did not want to duplicate it in each place.